cmd/snapd/main.go

```diff
@@ -91,7 +91,7 @@ func runWatchdog(d *daemon.Daemon) (*time.Ticker, error) {
 				//       replies with valid data
 				systemd.SdNotify("WATCHDOG=1")
 			case <-d.Dying():
-				break
+				return
```

Comment:
Is that something we should back port to stable? CC @mvo5

Reply:
I think cherry-picking this makes sense